### ISSUE TITLE
[Feature Request]: Add gz reader/writer to pcap files

### ISSUE THREAD

**Issue by felixzng** [seladb/PcapPlusPlus#128]
Would it be possible to add this functionality

**Comment by seladb** [comment]
What do you mean by gz reader/writer? 
I can think of various different options:
 - pcap files that are compressed with gz that you want to read/write
 - pcap files that contain packets and those packets contain files compressed with gz. In that case you first need to understand what is the protocol storing those files, decode that protocol and then decompress the file content

**Comment by felixzng** [comment]
I was thinking of pcap files that are compressed with gz

**Comment by seladb** [comment]
So just to make sure: the feature request is reading gz pcap files?
Why don't you extract them first the then read them with PcapPlusPlus?

**Comment by felixzng** [comment]
Right, I would like to read gzipped pcap file. Decompressing pcap files first would be time consuming and will require fair amount of extra space.  I have many terabytes of compressed pcaps.

**Comment by seladb** [comment]
I'm not sure this should be part of PcapPlusPlus as it has nothing to do with reading, analyzing or writing network packets. Arguably other people may request reading/writing pcap from/to a DB, etc.
Having said that, if you are willing to write the code and create a PR, I'd be happy to merge it into the main branch.
Also, we should consider the dependency on the gzip library (which currently doesn't exist)

**Comment by felixzng** [comment]
OK thx

**Comment by seladb** [comment]
Thanks! Please let me know if we should keep this issue open

**Comment by felixzng** [comment]
I think it's OK to close it. thx
